### DISCUSSION TITLE
Large differences in losses' magnitudes

### DISCUSSION THREAD

**Discussion by dang-trinh156** [lululxvi/deepxde#1294] [Q&A]
Hi Prof. Lulu and users,
I'm trying to apply PINNs in reservoir simulation, and here's overall PDEs for 3 phases in 1D (these equations are derived from Basic Applied Reservoir Simulation of Eterkin et al.)
![image](https://user-images.githubusercontent.com/124559501/236448868-cde26ed0-9331-4a26-8718-33b9597e814b.png)
My specific problem is solving SPE1 case. However, inherently, the result of Oil PDE is around 1e-5, Water PDE is around 1e-9, and Gas PDE is around 10. The initial loss is around 1e+3 (as my initial Pressure is around 6000). Assume that these losses are calculated in MAE.
Therefore, I've tried adaptive weight loss, but as oil and water PDEs are too low to update the weights in losses so that it's not working in this case. Another solution is that putting fixed weights for losses (1e+5 for Oil PDE, 1e+9 for Water PDE), but I still get wrong results. Is there any way to tackle this problem? 
P/s: my inputs are x (in ft) and t (time in days). My outputs are Pressure (should be more than 6000) and Sw (water saturation in range [0,1]) and Sg (gas saturation in range [0,1])
Thank you for your answers!

**Comment by lululxvi** [comment]
See FAQ Q: I failed to train the network or get the right solution, e.g., large training loss, unbalanced losses.
